### PR TITLE
Issue #493: Piwik Template integration

### PR DESCRIPTION
**Pull request by pascalchevrel**
- add piwik_url and piwik_id variables to config.ini
- integrate js only in production mode, if the settings exist and are valid

### CODE REVIEW

File: app/inc/constants.php [hunk 1]
```diff
@@ -28,7 +28,8 @@
 }
 
 // Special modes for the app
-define('DEBUG', (strstr(VERSION, 'dev') || isset($_GET['debug'])) ? true : false);
+define('DEBUG',     strstr(VERSION, 'dev') || isset($_GET['debug']));
+define('LOCAL_DEV', isset($server_config['dev']) && $server_config['dev']);
```
Comment (flodolo) [inline app/inc/constants.php]: I'm lost, where should `$server_config['dev']` come from?

Comment (pascalchevrel) [inline app/inc/constants.php]: https://github.com/mozfr/transvision/blob/master/app/config/config.ini-dev#L27

File: app/views/templates/piwik.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  _paq.push(['enableLinkTracking']);
+  (function() {
+    var u="<?=$server_config['piwik_url']?>";
+    _paq.push(['setTrackerUrl', u+'piwik.php']);
+    _paq.push(['setSiteId', <?=$server_config['piwik_id']?>]);
+    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
+    g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'piwik.js'; s.parentNode.insertBefore(g,s);
+  })();
+</script>
+<noscript><p><img src="<?=$server_config['piwik_url']?>piwik.php?idsite=<?=$server_config['piwik_id']?>" style="border:0;" alt="" /></p></noscript>
```
Comment (flodolo) [inline app/views/templates/piwik.php]: Should you have a '/` before `piwik.php`? Anyway worth a note in the comment in config.ini-dist, if the URL has to have a trailing slash or not.

Comment (pascalchevrel) [inline app/views/templates/piwik.php]: it needs a trailing slash, will add a note

File: app/inc/constants.php (hunk 1 above)
Comment (flodolo) [inline app/inc/constants.php]: Wow, never noticed it. Are we already using it somewhere? Can't find anything in the code.

Can you add `dev=false`to config.ini-dist? I also wonder if we should just use one system to determine if it's dev or not (I always considered the version to be the trigger, but I like a specific parameter a lot more).

**Comment by flodolo** [comment]
Besides my confusion about `dev` in ```config.ini``` (which can be fixed/discussed somewhere else), this should be good to merge.

File: app/inc/constants.php (hunk 1 above)
Comment (pascalchevrel) [inline app/inc/constants.php]: we use it in the dev-setup script to install or not a snapshot:
https://github.com/mozfr/transvision/blob/master/app/scripts/dev-setup.sh#L79

Yes, I can add dev=false to -dist

There is a difference between a dev setup and a dev version (in that context it means beta), maybe we should use beta instead but I would prefer that to be part of a separate issue.

Comment (pascalchevrel) [inline app/inc/constants.php]: actually, dev=false is already in config.ini-dist:
https://github.com/mozfr/transvision/blob/master/app/config/config.ini-dist#L32

**Pull request merged**